shared/src/main/scala/tastyquery/reader/TreeUnpickler.scala

```diff
@@ -996,6 +996,9 @@ class TreeUnpickler(
         underlying
     case tag if (isConstantTag(tag)) =>
       ConstantType(readConstant)
+    case RECtype =>
+      reader.readByte()
+      readType
```

Comment:
This should definitely interact with `recursiveTypeAtAddr` with "itself" before diving into `readType`. That means building a `RecType` with a lambda that passes itself. Similarly to how it's done in dotty.